examples/showcase/post-processing/bloom.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Unreal Bloom Effect
+ * Implementation for A-Frame
+ * Code modified from Akbartus's UnrealBloomPass.js
+ * https://github.com/akbartus/A-Frame-Component-Postprocessing
+ */
+
+import AFRAME from "aframe";
+import { EffectComposer } from "three/addons/postprocessing/EffectComposer.js";
+import { RenderPass } from "three/addons/postprocessing/RenderPass.js";
+import { UnrealBloomPass } from "three/addons/postprocessing/UnrealBloomPass.js";
+
+/////////////////////////////////////
+// A-FRAME COMPONENT: BLOOM        //
+/////////////////////////////////////
+AFRAME.registerComponent("bloom", {
+  schema: {
+    threshold: {
+      type: "number",
+      default: 0,
+    },
+    strength: {
+      type: "number",
+      default: 0.4,
+    },
+    radius: {
+      type: "number",
+      default: 0,
+    },
+  },
+  init: function () {
+    this.bind();
+  },
+  update: function (oldData) {
+    this.evaluateEffect();
+  },
+  evaluateEffect: function () {
+    this.scene = this.el.object3D;
+    this.renderer = this.el.renderer;
+    this.camera = this.el.camera;
+    this.composer = new EffectComposer(this.renderer);
+    const renderScene = new RenderPass(this.scene, this.camera);
+    this.composer.addPass(renderScene);
+
+    // creating bloom pass    
+    const bloomPass = new UnrealBloomPass(
+      new THREE.Vector2(
+        this.renderer.domElement.clientWidth,
+        this.renderer.domElement.clientHeight
+      ),
+      1.5,
+      0.4,
+      0.85
+    );
+
+    bloomPass.fsQuad.render = function (renderer) {
+      // Disable XR projection for fullscreen effects
+      // https://github.com/mrdoob/three.js/pull/18846
+      const xrEnabled = renderer.xr.enabled;
+      const _camera = new THREE.OrthographicCamera(- 1, 1, 1, - 1, 0, 1);
+      renderer.xr.enabled = false;
+      renderer.render(this._mesh, _camera);
+      renderer.xr.enabled = xrEnabled;
+    }
+
+    bloomPass.threshold = this.data.threshold;
+    bloomPass.strength = this.data.strength;
+    bloomPass.radius = this.data.radius;
+    this.composer.addPass(bloomPass);
+  },
+
+  bind: function () {
+    const render = this.el.renderer.render;
+    const system = this;
+    let isInsideComposerRender = false;
+
+    this.el.renderer.render = function () {
+
+      if (isInsideComposerRender) {
+        render.apply(this, arguments);
+      } else {
+        isInsideComposerRender = true;
+        system.composer.render(system.dt);
```

Comment:
If I'm not mistaken, `system.dt` is undefined now we removed the assignment in tick, that was a hack to get the value because the tick is called before renderer.render
https://github.com/aframevr/aframe/blob/c2684072e3eb332ea159b9c29bab8d98ddfdd53d/src/core/scene/a-scene.js#L653-L664
You should be able to use `self.el.sceneEl.delta` I think here (verify with a console.log).
And change `const self = this` instead of `const system = this;` above.

I don't quite understand that if (isInsideComposerRender), isInsideComposerRender is always false here, right?

Reply:
I did point out the `system` -> `self` rename and that the delta time should be retrieved from the scene instead. Note that it should be divided by 1000 as `EffectComposer` expects the value in seconds, not milliseconds. @enzofrancescaHM Did some of your changes get lost when moving the files around?

As for `isInsideComposerRender`; the effect composer will make calls to `renderer.render()` itself, in which case the original `render` method should be used. This flag is a crude way to distinguish between the outermost invocation (called by the a-scene) and inner invocation called while inside the render method of the composer.